app/lib/account/auth_provider.dart

```diff
@@ -41,6 +47,17 @@ abstract class AuthProvider {
   /// Returns the profile information of a given access token.
   Future<AccountProfile?> getAccountProfile(String? accessToken);
 
+  /// Returns the URL that the user should visit for authentication.
+  Future<Uri> getOauthAuthenticationUrl({
+    required String state,
+    required String nonce,
+  });
+
+  /// Returns the OAuth2 profile information using the authentication [code].
+  Future<AuthResult?> tryAuthenticateOauthCode({
+    required String code,
+  });
```

Comment:
Why not stick the nonce back in here?

So that the verification of it happens here? The `getOauthAuthenticationUrl` takes a `nonce`.